pkg/factory/risingwave_object_factory.go

```diff
@@ -1336,6 +1515,53 @@ func (f *RisingWaveObjectFactory) NewComputeStatefulSet(group string, podTemplat
 	return mustSetControllerReference(f.risingwave, computeStatefulSet, f.scheme)
 }
 
+func (f *RisingWaveObjectFactory) NewComputeAdvancedStatefulSet(group string, podTemplates map[string]risingwavev1alpha1.RisingWavePodTemplate) *kruiseappsv1beta1.StatefulSet {
+	componentGroup := buildComputeGroup(
+		f.risingwave.Spec.Global.Replicas.Compute,
+		&f.risingwave.Spec.Global.RisingWaveComponentGroupTemplate,
+		group,
+		f.risingwave.Spec.Components.Compute.Groups,
+	)
+	if componentGroup == nil {
+		return nil
+	}
+	restartAt := f.risingwave.Spec.Components.Compute.RestartAt
+	pvcTemplates := f.risingwave.Spec.Storages.PVCTemplates
+
+	podTemplate := f.buildPodTemplate(consts.ComponentCompute, group, podTemplates, &componentGroup.RisingWaveComponentGroupTemplate, restartAt)
+
+	// Readiness gate InPlaceUpdateReady required for advanced statefulset
+	podTemplate.Spec.ReadinessGates = append(podTemplate.Spec.ReadinessGates, corev1.PodReadinessGate{
+		ConditionType: "InPlaceUpdateReady",
```

Comment:
Is there a constant we can refer to in the openkruise api package?

Reply:
have added it in the latest update